src/main/java/duke/Task.java

```diff
@@ -0,0 +1,48 @@
+package duke;
+
+import java.time.LocalDateTime;
+
+public class Task {
```

Comment:
Should `Task` be an abstract class, considering that it shouldn't be initialized on its own? I noticed that you have overwritten a few methods like mark() in your child-classes, maybe those can be the abstract methods!